docs/js/features/odata/use-vdm.md

```diff
@@ -0,0 +1,280 @@
+---
+id: use-typed-odata-v4-client-for-js
+title: Use the type-safe client to consume OData Services for JS/TS
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Use the type-safe client to consume OData Services for JS/TS
+description: Use SAP Cloud SDK for JavaScript to build and run OData requests in a typesafe way.
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- consume
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Build and execute OData Requests with the typed OData client
+
+The typed OData client allows to build type-safe OData requests for a given service. All requests that are build through the typed OData client are not only _syntactically valid_ but also _semantically valid_.
+
+## Using the Fluent API ##
+
+The typed OData client consists of a service and a controller. The service mirrors the API provided by the OData service and serves as an entry point for creating requests. It provides a builder which allows for adding further parameters in a fluent way.
+
+First, import the generated service:
+
+```ts
+import { <YourData> } from '<outputDir>/<npmPackageName>';
+```
+
+Where `<outputDir>` is the directory you specified when generating the service and `<npmPackageName>` was specified inside the `serviceMapping.json` file which is inside your input directory.
+On an abstract level requests are generally build up according to the following pattern:
+
+```ts
+return <YourData>.requestBuilder()
+    .withParameter()
+    .execute({
+        url: '<yourURL>'
+    });
+```
+
+- _requestBuilder_ to construct requests for operations.
+- _withParameter_ corresponds to:
+    - OData query parameters e.g. `filter` or `orderby`
+    - Or other modifiers like custom headers
+- Which OData parameters are available depends on the operation. For example when updating entities the `$filter` parameter is not available.
+
+Below different OData features are documented using the [Business Partner Service](https://api.sap.com/api/API_BUSINESS_PARTNER/resource) on S/4HANA as an example. The following code snippets assume that an instance of this service is set up:
+
+```ts
+constructor(private readonly businessPartnerService: BusinessPartnerService) {}
+```
+
+:::tip Structure your code
+All of the following code can be organised in service files, e.g., `YourData.service.ts`
+:::
```

Comment:
What is this? Sounds like Java?

Reply:
I used the constructor inside my controller to instantiate my service. Let me know if this is useful or not.
And the tip is for the user to know where to put his code, but maybe I shouldn't talk about the notion of service and controller, it seems to bring up more confusion.

Reply:
Yes, I think this is more nestJs related and does not really refer to the odata client itself.